### ISSUE TITLE
[Conv3D] `ttnn.experimental.conv3d` returns near-zero PCC for non-overlapping low-channel patchify

### ISSUE THREAD

**Issue by sonalibaskaran2499** [tenstorrent/tt-metal#47316]
## Describe the bug

`ttnn.experimental.conv3d` produces results that are **uncorrelated** with the reference when the convolution is a *non-overlapping patchify* (`stride == kernel_size`) over a **small input-channel count** (`C_in = 3`).

The exact case is Qwen2.5-VL's vision **patch embedding**:

```
in_channels  = 3
out_channels = 1280
kernel_size  = stride = [2, 14, 14]   (D, H, W)
padding      = [0, 0, 0]
groups       = 1
bias         = none
```

Reference (`torch.nn.functional.conv3d`) vs `ttnn.experimental.conv3d`:

| op | PCC |
|---|---|
| `ttnn.experimental.conv3d` (C_in = 3) | **≈ 0.0048** (near-zero / uncorrelated) |
| `ttnn.experimental.conv3d` (C_in = 32) | passing |

Because this op is the **very first** operation in the model, the broken numerics propagate everywhere: full-model Qwen2.5-VL-3B PCC collapses from ~0.90 to **0.24**.

## Root cause 

`ttnn.conv3d` lowers to **vol2col → tilize → matmul**. With only 3 input channels, the channel dimension is padded out to a 32-wide tile. The `vol2col` activation packing and the separately-prepared weight layout appear to **disagree on where the 3 real channels sit inside the padded 32-wide tile**, so the matmul contracts mismatched activation/weight pairs. This is consistent with the result being correct for `C_in = 32` (no channel padding) and uncorrelated for `C_in = 3`.

Suspect code paths:
- `ttnn/cpp/ttnn/operations/experimental/conv3d/device/kernels/reader_vol2col.cpp` (channel padding into tiles)
- `ttnn/cpp/ttnn/operations/experimental/conv3d/device/kernels/compute.cpp` (`matmul_block`)
- `ttnn/cpp/ttnn/operations/experimental/conv3d/device/prepare_conv3d_weights.cpp` (weight repacking)

## Steps to reproduce

The script runs the failing 3-channel case and a passing 32-channel control so the channel-padding dependence is obvious.

```python
import torch
import ttnn

# Qwen2.5-VL vision patch embedding: conv3d with stride == kernel_size, C_in = 3.
IN_CHANNELS = 3
OUT_CHANNELS = 256                       # kept small so the program fits L1
TEMPORAL_PATCH_SIZE, PATCH_SIZE = 2, 14
KERNEL = (TEMPORAL_PATCH_SIZE, PATCH_SIZE, PATCH_SIZE)
ALIGN = 32


def pcc(a, b):
    a = a.flatten().to(torch.float64)
    b = b.flatten().to(torch.float64)
    if torch.allclose(a, b):
        return 1.0
    return torch.corrcoef(torch.stack([a, b]))[0, 1].item()


def run(prep_c_in_block, conv_c_in_block, label, patches=8):
    stride, padding = KERNEL, (0, 0, 0)
    torch.manual_seed(0)
    inp = torch.randn(patches, IN_CHANNELS, *KERNEL)
    wgt = torch.randn(OUT_CHANNELS, IN_CHANNELS, *KERNEL)

    ref = torch.nn.functional.conv3d(inp, wgt, None, stride, padding)
    ref = ref.reshape(patches, OUT_CHANNELS)

    # input -> [N, D, H, W, C] with the 3 real channels zero-padded up to the 32 alignment
    c_pad = (ALIGN - IN_CHANNELS % ALIGN) % ALIGN
    inp_ndhwc = torch.nn.functional.pad(inp.permute(0, 2, 3, 4, 1).contiguous(), (0, c_pad))
    tt_input = ttnn.from_torch(
        inp_ndhwc, dtype=ttnn.bfloat16, layout=ttnn.ROW_MAJOR_LAYOUT, device=device
    )

    # weight prepared separately, PyTorch-native [C_out, C_in, kD, kH, kW] on host
    tt_weight_host = ttnn.from_torch(wgt, dtype=ttnn.bfloat16, layout=ttnn.ROW_MAJOR_LAYOUT)
    tt_weight = ttnn.experimental.prepare_conv3d_weights(
        weight_tensor=tt_weight_host, groups=1, C_in_block=prep_c_in_block, alignment=ALIGN, device=device
    )

    config = ttnn.Conv3dConfig(
        weights_dtype=ttnn.bfloat16,
        output_layout=ttnn.ROW_MAJOR_LAYOUT,
        C_out_block=32,
        C_in_block=conv_c_in_block,
        compute_with_storage_grid_size=device.compute_with_storage_grid_size(),
    )

    out = ttnn.experimental.conv3d(
        input_tensor=tt_input,
        weight_tensor=tt_weight,
        device=device,
        config=config,
        dtype=ttnn.bfloat16,
        output_channels=OUT_CHANNELS,
        kernel_size=KERNEL,
        stride=stride,
        padding=padding,
        padding_mode="zeros",
        groups=1,
    )
    out = ttnn.to_torch(out).reshape(patches, OUT_CHANNELS).to(torch.float32)

    p = pcc(ref, out)
    print(f"{label:32s} prepare C_in_block={prep_c_in_block:2d}  conv C_in_block={conv_c_in_block:2d}  "
          f"PCC={p:.4f}  {'FAIL' if p < 0.99 else 'pass'}")


device = ttnn.open_device(device_id=0)
try:
    run(16, 16, "matched")            # weight blocking == compute blocking -> correct
    run(32, 16, "MISMATCH (= tt-mlir)")  # prep default 32 vs conv default 16 -> near-zero PCC
finally:
    ttnn.close_device(device)
```
